### PR TITLE
Sync and add 'padding-inline' UA stylehseet rules for `mfrac` in mathml.css

### PR DESCRIPTION
**Pull request by Ahmad-S792**
#### 612aa962c630f31a2e5feca8b0df7b4b344a9eab
<pre>
Sync and add &apos;padding-inline&apos; UA stylehseet rules for `mfrac` in mathml.css

<a href="https://bugs.webkit.org/show_bug.cgi?id=262789">https://bugs.webkit.org/show_bug.cgi?id=262789</a>
<a href="https://rdar.apple.com/problem/116922348">rdar://problem/116922348</a>

Reviewed by Frédéric Wang.

This patch aligns WebKit with Web-Specification [1] by adding &apos;padding-inline-*&apos; for `mfrac`.

[1] <a href="https://w3c.github.io/mathml-core/#user-agent-stylesheet">https://w3c.github.io/mathml-core/#user-agent-stylesheet</a>

* Source/WebCore/css/mathml.css:
* LayoutTests/platform/mac/mathml/presentation/roots-expected.txt: Rebaselined
* LayoutTests/platform/glib/mathml/presentation/roots-expected.txt: Rebaselined
* LayoutTests/platform/gtk/mathml/presentation/roots-expected.png: Rebaselined
* LayoutTests/platform/ios/mathml/presentation/roots-expected.png: Rebaselined
* LayoutTests/platform/ios/mathml/presentation/roots-expected.txt: Rebaselined
* LayoutTests/platform/mac-monterey/mathml/presentation/roots-expected.txt: Rebaselined
* LayoutTests/imported/w3c/web-platform-tests/mathml/presentation-markup/fractions/default-mfrac-padding-style.html: Add Test Case
* LayoutTests/imported/w3c/web-platform-tests/mathml/presentation-markup/fractions/default-mfrac-padding-style-expected.txt: Add Test Case Expectation

Canonical link: <a href="https://commits.webkit.org/271337@main">https://commits.webkit.org/271337@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/d76eb5c5140e069829f182a55fa3b5916cc278f6

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/27970 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/6608 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/29271 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/30500 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/25514 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/28465 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/8602 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/3999 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/25278 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/28237 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/5361 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/24025 "Passed tests") | [❌ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/4618 "Found 1 new API test failure: /WPE/TestWebKitWebView:/webkit/WebKitWebView/notification (failure)") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/4800 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/25031 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/31189 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/25573 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/25479 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/31057 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/4807 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/2970 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/28869 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/6341 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/24798 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/6726 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/5252 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/5292 "Built successfully") | | |

### CODE REVIEW

**Comment by Ahmad-S792** [comment]
My main aim with these small alignments are to work toward 'mathml-core' without doing disruption changes and it might not give us any 'WPT' win but it will still helps us doing small incremental improvements.

**Comment by fred-wang** [comment]
This looks OK, but I'm curious whether that's causing any visual improvement?

Basic test case is:

```
    <math>
      <mfrac>
        <mn>1</mn>
        <mn>2</mn>
      </mfrac>
      <mfrac>
        <mn>3</mn>
        <mn>4</mn>
      </mfrac>
    </math>
```

which should render as two fractions 1/2 and 3/4 with a gap between them. Does WebKit add a gap already? Does the patch improves it?

**Review by fred-wang — APPROVED** [review]

File: Source/WebCore/css/mathml.css [hunk 1]
```diff
@@ -155,6 +155,12 @@ ms, mtext, mi, mn, mo, annotation, mtd {
     white-space: nowrap !important;
 }
 
+/* Fractions */
+mfrac {
+    padding-inline-start: 1px;
+    padding-inline-end: 1px;
```
Comment (nt1m) [inline Source/WebCore/css/mathml.css]:
```suggestion
    padding-inline: 1px;
```

This can be shortened

Comment (Ahmad-S792) [reply Source/WebCore/css/mathml.css]: Done in 3f75cc7.

**Comment by fred-wang** [comment]
> It seems that it does not improve it. Test (based on above) :

Thanks. So it looks we will need to fix padding support for MathML renderers to make that work.

**Comment by Ahmad-S792** [comment]
> > It seems that it does not improve it. Test (based on above) :
> 
> 
> 
> Thanks. So it looks we will need to fix padding support for MathML renderers to make that work.

Yes, although I noticed that this PR also improves spacing in fractions and makes them readable at least more spaced out. So I think it is still OK to land this. Any thoughts?

**Comment by fred-wang** [comment]
> > > It seems that it does not improve it. Test (based on above) :
> > 
> > 
> > Thanks. So it looks we will need to fix padding support for MathML renderers to make that work.
> 
> Yes, although I noticed that this PR also improves spacing in fractions and makes them readable at least more spaced out. So I think it is still OK to land this. Any thoughts?

Interesting, so maybe the padding is taken into account somewhere else. Anyway,  I think it's fine to land it as is. However, I believe there should be a WPT test for that... Maybe existing tests only check the layout boxes, not the actual default style.

Can you please write a test in `/mathml/presentation-markup/fractions` similar to `mathml/presentation-markup/mrow/merror-001.html`? IIUC, your patch should make a change for `getComputedStyle` but for `getBoundingClientRect`. Thanks

**Comment by Ahmad-S792** [comment]
> > > > It seems that it does not improve it. Test (based on above) :
> > > 
> > > 
> > > Thanks. So it looks we will need to fix padding support for MathML renderers to make that work.
> > 
> > 
> > Yes, although I noticed that this PR also improves spacing in fractions and makes them readable at least more spaced out. So I think it is still OK to land this. Any thoughts?
> 
> Interesting, so maybe the padding is taken into account somewhere else. Anyway, I think it's fine to land it as is. However, I believe there should be a WPT test for that... Maybe existing tests only check the layout boxes, not the actual default style.
> 
> Can you please write a test in `/mathml/presentation-markup/fractions` similar to `mathml/presentation-markup/mrow/merror-001.html`? IIUC, your patch should make a change for `getComputedStyle` but not for `getBoundingClientRect`. Thanks

@fred-wang - Thanks for guidance. Just added one test case, appreciate if you can review and if it is OK, I am happy to do PR on WPT as well. I don't know how to use 'export' script. :-)

**Comment by fred-wang** [comment]
@Ahmad-S792 Thanks the WPT test looks good to me. I'm not quite up-to-date about WebKit's policy for WPT tests, but IIRC we need first to add the test in upstream WPT repo before adding it to WebKit's own copy. Whether that's done via the export script or not is up to you, but please don't merge the WebKit PR until the test is in upstream WPT.

**Comment by Ahmad-S792** [comment]
> @Ahmad-S792 Thanks the WPT test looks good to me. I'm not quite up-to-date about WebKit's policy for WPT tests, but IIRC we need first to add the test in upstream WPT repo before adding it to WebKit's own copy. Whether that's done via the export script or not is up to you, but please don't merge the WebKit PR until the test is in upstream WPT.

I think WebKit policy is that we can land this without first uploading to WPT and then we can use 'export' script to upload to WPT or do manual PR later. Although, I have just did PR on WPT:

https://github.com/web-platform-tests/wpt/pull/43449 - tagged yourself for review.

**Pull request merged**